snuba/subscriptions/consumer.py

```diff
@@ -122,6 +122,31 @@ def poll(self, timeout: Optional[float] = None) -> Optional[Message[Tick]]:
 
         return result
 
+    def tell(self) -> Mapping[Partition, int]:
+        # If there is no previous message for a partition, return the current
+        # consumer offset, otherwise return the previous message offset (which
+        # will be the next offset returned for that partition) to make the
+        # behavior of the consumer consistent with what would typically be
+        # expected by the caller.
+        return {
+            partition: (
+                self.__previous_messages[partition].offset
+                if partition in self.__previous_messages
+                else offset
+            )
+            for partition, offset in self.__consumer.tell().items()
+        }
+
+    def seek(self, offsets: Mapping[Partition, int]) -> None:
+        if offsets.keys() - self.__consumer.tell().keys():
+            raise ConsumerError("cannot seek on unassigned partitions")
```

Comment:
This gets duplicated fairly often between the various consumer implementations, but I'm not sure abstracting it is worth the effort or complexity cost. I'd rather move towards having a test suite that can be parameterized and run against each different consumer implementation without having to rewrite the same tests over and over.